tests/units/units.cpp

```diff
@@ -1471,3 +1471,198 @@ TEST(Units, compareEquivalentSameSizeButDifferentExponent)
     EXPECT_FALSE(libcellml::Units::equivalent(u1, u2));
     EXPECT_FALSE(libcellml::Units::equivalent(u2, u1));
 }
+
+TEST(Units, isBaseUnitsImported)
+{
+    libcellml::ModelPtr model = libcellml::Model::create();
+    model->setName("model");
+
+    // Making a base unit
+    libcellml::UnitsPtr u = libcellml::Units::create();
+    u->setName("u");
+
+    libcellml::ImportSourcePtr import = libcellml::ImportSource::create();
+    import->setUrl("I_am_a_url");
+
+    model->addUnits(u);
+
+    import->setModel(model);
```

Comment:
You have made a circular reference here, no way a shared pointer can release the memory thus the memory leak.

Reply:
The model that gets set to the import is the model that has the units you are referencing.

Reply:
So instead, to avoid the circular reference, we would create two units with the same name, one in the `model` and one outside the `model`, then checking?

Reply:
You have to create a whole separate model that has the units defined that another model is going to import from see [tests/model/model_flattening.cpp](https://github.com/hsorby/libcellml/blob/b83d125346d04323e9482754ac937a556da860f8/tests/model/model_flattening.cpp#L69) for an example.